packages/@aws-cdk/aws-lambda/lib/log-retention-provider/index.ts

```diff
@@ -3,14 +3,17 @@
 // eslint-disable-next-line import/no-extraneous-dependencies
 import * as AWS from 'aws-sdk';
 
+// Ensure we do not run into throttling issues when deploying stack(s) with a lot of Lambdas.
+const retryOptions = { maxRetries: 6, retryDelayOptions: { base: 300 }};
```

Comment:
How about making this configurable here - https://github.com/aws/aws-cdk/blob/fd54a17a82605ac1301e5776aa68f03bbfb63910/packages/%40aws-cdk/aws-lambda/lib/log-retention.ts#L12 - and leave the default as the SDK's default?

Reply:
Ok, that makes sense. Do you mean something like this:

1. Add retry options to `LogRetentionProps`:
```
export interface LogRetentionProps {
  readonly logGroupName: string;
  readonly retention: logs.RetentionDays;
  readonly role?: iam.IRole;
  readonly logGroupRetryOptions?:LogGroupRetryOptions;
}

export interface LogGroupRetryOptions {
  readonly maxRetries?: number;
  readonly retryDelayOptions?: {
     readonly base?: number
     readonly customBackoff?: (retryCount: number, err?: Error) => number
  }
}
```

2. Allow the retry options to be configured in the Lambda by adding  the `LogGroupRetryOptions` property also in `FunctionOptions`.

3. Use the retry options in the log retention provider Lambda.

Reply:
You don't need to do #2. 

Once defined in `LogRetentionProps`, pass it into the [`properties` key](https://github.com/aws/aws-cdk/blob/fd54a17a82605ac1301e5776aa68f03bbfb63910/packages/%40aws-cdk/aws-lambda/lib/log-retention.ts#L69) and it should be available in the provider lambda. Just like the other values under `properties`.

Reply:
But I guess the goal here is to make the retry options configurable via the Lambda props? So when a CDK users creates a Lambda, they can also specify the retry options.

The `logRetention` and `logRetentionRole` props are also part of `FunctionOptions` to [enable](https://github.com/aws/aws-cdk/blob/fd54a17a82605ac1301e5776aa68f03bbfb63910/packages/%40aws-cdk/aws-lambda/lib/function.ts#L218) something similar.

Reply:
@nija-at Just checking, does my previous comment makes sense? I'd like to continue and obviously create a wonderful PR to fix the issue 🙂

Reply:
Yes, something like that makes sense. Thanks for explaining.

Reply:
Made the changes, please check the PR description for additional details on implementation choices.